### ISSUE TITLE
Integrate user query with the UI

### ISSUE THREAD

**Issue by diegoTech14** [safetrustcr/Frontend#149]
# 📝 Implement User Data Query  

## 🔍 Description  
Implement a GraphQL query to fetch user profile data and their associated wallet information to populate the Profile Settings page. This follows the existing query structure from PR#71.

---

## 🔄 Implementation Steps  
1. Implement the GraphQL query in the Profile Settings component.  
2. Add loading and error handling states.  
3. Connect the returned data to the Profile Settings UI components.  
4. Implement a proper data caching strategy using Apollo Client.  

---

## 📊 Query Structure  
```graphql
query GetUserProfileData($userId: uuid!) {
  users(where: { id: { _eq: $userId } }) {
    id
    email
    first_name
    last_name
    created_at
    updated_at
  }
  user_wallets(where: { user_id: { _eq: $userId } }) {
    id
    wallet_address
    is_primary
    created_at
  }
}
```

## 📋 Expected Types
```
interface UserProfileData {
  users: {
    id: string;
    email: string;
    first_name: string | null;
    last_name: string | null;
    created_at: string;
    updated_at: string;
  }[];
  user_wallets: {
    id: string;
    wallet_address: string;
    is_primary: boolean;
    created_at: string;
  }[];
}

```

## 👌🏼 Expected Response Example
```
{
  "data": {
    "users": [{
      "id": "37e4aaca-6812-473c-adfb-5f6578148f2d",
      "email": "user@example.com",
      "first_name": "John",
      "last_name": "Doe",
      "created_at": "2024-01-01T00:00:00Z",
      "updated_at": "2024-01-01T00:00:00Z"
    }],
    "user_wallets": [{
      "id": "wallet-uuid",
      "wallet_address": "0x123...",
      "is_primary": true,
      "created_at": "2024-01-01T00:00:00Z"
    }]
  }
}

```

## 🔧 Error Handling  
Implement handling for the following cases:  
- User not found  
- Network errors  
- Empty wallet list  
- Invalid wallet addresses  

---

## 💾 Caching Considerations  
- Use Apollo Client's cache for storing user data.  
- Implement cache invalidation strategy for wallet updates.  
- Consider optimistic updates for wallet changes.  

---

## 🖥️ File Structure  
The query should be implemented in:  

```bash
📁 src/features/profile/hooks/useUserProfile.ts
```

## 🕹️ Loading & Error States  
- **Loading State**: Display appropriate skeleton components.  
- **Error State**: Show user-friendly error messages with retry options.  

---

## 📚 Additional Information  
- Ensure the query fetches both user data and wallet information accurately.  
- Follow the structure and patterns from the referenced PR.  
- Include proper retry logic for failed queries.  

---

Let me know if any further refinements are needed! 🚀

**Comment by hoangkianh** [comment]
I'd love to give this a go. The ETA is 1 - 2 days

**Comment by Jagadeeshftw** [comment]
Hi @diegoTech14,
I would like to work on this issue! I have experience implementing GraphQL queries and integrating data with UI components.
Please assign it to me. I’ll make sure to:

Query the backend to fetch user and wallet data.
Integrate the data with the profile settings UI as expected.
Test the query and UI to ensure it works seamlessly.   ETA: less than 1 day

**Comment by jahrulezfrancis** [comment]
I’d like to take on this task and deliver the user data query as required. I’ll handle the GraphQL implementation, ensure the data flows into the profile settings UI, and verify everything works smoothly. My ETA is 24 to 36 hours

**Comment by Agbeleshe** [comment]
Can I be assigned to this?
I am a frontend developer
ETA:1 day

**Comment by davedumto** [comment]
Hi,

My name is David, and I am a frontend developer skilled in JavaScript, TypeScript, React.js, Next.js, and Apollo GraphQL. I will implement the GraphQL query to fetch user data, including wallet addresses, and use it to populate the profile settings UI.

I’ll ensure the query is correctly structured, data is retrieved seamlessly, and the profile settings UI is fed with accurate user information. ETA: 2 days.

**Comment by olisaagbafor** [comment]
Can I be assigned to this?

**Comment by akintewe** [comment]
Can I take care of this issue?

**Comment by Crosstons** [comment]
I'd like to take this issue.

**Comment by OWK50GA** [comment]
##My Approach To Solving This Issue:
I will make the graphQL query, making sure it requests for all the pieces of information required for the profile details as provided above. 
I will use different TypeScript techniques to make sure the data arrived well and without error. 
Then I will go into the appropriate component and render. 

Expect my first PR within 24 hours of assignment.

**Comment by ryzen-xp** [comment]
Can I start working on this?

**Comment by mimisavage** [comment]
Would love to tackle this!

**Comment by martinvibes** [comment]
may i handle this task?
hey i'm a frontend dev and blockchain dev and i would like to work on this issue 
kindly assign :)

**Comment by emarc99** [comment]
Is it okay if I tackle this?

**Comment by Prosper1218** [comment]
can i work on this for you 😊.
I'll make my first PR within 24hrs

**Comment by FrankiePower** [comment]
I am Super Franky, a software developer with 3+ years of experience, and I have worked with frontend technologies like React, NextJS, JavaScript/TypeScript. I am also proficient in Cairo for smart contract development. I'm a recent graduate of Web3bridge Africa, the foremost and premier blockchain developer community. I have made contributions to many open-source projects, and I look forward to contributing to this one. My approach to this will be to:

1. Review the existing query structure implemented to understand how the current queries are structured and how the data is being fetched.

2. Implement a GraphQL query that will query the backend to fetch the necessary user data (e.g., username, email, preferences) for the profile settings.

3. Test the query to ensure the correct data is returned from the backend and that the response is in the expected format.

4. Feed the profile settings UI with the user data received from the query. Ensure that the UI components (e.g., text fields, dropdowns, checkboxes) are populated with the correct values.

ETA: 3 days
